cluster-http/src/main/scala/akka/management/cluster/scaladsl/ClusterReadinessCheck.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.management.cluster.scaladsl
```

Comment:
One important question you raised yesterday was about if this ClusterReadinessCheck should be in `akka-management-cluster-http` or in `akka-cluster` itself. As mentioned elsewhere we might add more specific checks for Cluster Sharding and similar.

I'm on the fence here.

On one hand it's nice to define the checks in the library modules themselves, so that `akka-management-cluster-http` doesn't have to dependencies to a lot of things. It already has dependencies to `akka-cluster` and `cluster-sharding` so that is not a problem for the checks we know about currently.

On the other hand I think these health checks are rather Akka Management (for K8 and similar env) and therefore don't belong in core Akka modules.

We had a similar discussion about if the contact point probing should be moved to akka-cluster and using Akka remoting instead of http. We decided for now that we continue with the http probing. Perhaps this question is similar? For now we define the cluster health checks in `akka-management-cluster-http`?

Reply:
I think the same checks would be used in a traditional environment for the LoadBalancer health checks so as long as we make it easy to use them without bootstrap I think they can start their life in akka management.

The checks are agnostic of http but perhaps they can stay in `cluster-http` until someone requests to use them without http? 

I foresee us adding ones for `akka-persistence-cassandra` to check the connection to Cassandra (perhaps requiring an Extension to get hold of the cluster like Couchbase has)

If we add checks for Singleton, Sharding then this probably means creating a `akka-management-cluster-singleton` etc rather than polluting the classpath when using just cluster (not sure how often cluster is used without singleton/sharding).

Reply:
Leaving these in here for now